server/src/core/payment/transaction-service.ts

```diff
@@ -66,7 +66,8 @@ export class Transactions implements TransactionService {
   }
 
   async initiateDonation(user: User, args: InitiateDonationArgs): Promise<Transaction> {
-    validators.validatesInitiateDonation({ userId: user._id, amount: args.amount })
+    validators.validatesInitiateDonation({ userId: user._id, amount: args.amount });
+    args.amount = Math.floor(args.amount);
```

Comment:
I don't think it's a good idea to modify the input object. Would have been better to create a new variable `const amount = Math.floor(args.amount)` to avoid modifying the arguments.